framework/src/org/checkerframework/framework/util/element/ElementAnnotationUtil.java

```diff
@@ -78,7 +78,8 @@ public static void applyAllElementAnnotations(
      * understand them as type annotations. In particular, this allows the transition from Java 5
      * declaration annotations to Java 8 type annotations.
      *
-     * <p>There are some caveats to this: the interpretation for declaration and type annotations
+     * <p>If the annotation is one of the Checker Framework compatibility annotations, for example
+     * {@link org.checkerframework.checker.nullness.compatqual.NonNullDecl}, the interpretation
```

Comment:
Do we need `@NonNullDecl` any more?  Since the Checker Framework does not support Java 7 any more, I would think that we can eliminate the `@NonNullDecl` and all related annotations.  Then we don't need to talk about them, anywhere in the manual.